PostgreSQL Plugin

jlsuttles opened the pull request:
Adds the ability to save cypress log messages to a postgresql database.

jlsuttles committed:
Starting postgresql plugin with evan

jlsuttles committed:
Get timestamp, attributes, and tags saving to postgres

jlsuttles commented on plugin/postgresql_test.go:
```diff
@@ -0,0 +1,102 @@
+
+		db.On("Ping").Return(nil)
+		db.On("Exec", cCreateTable, []interface{}{}).Return(mock.Anything, nil)
+		db.On("Exec", cEnableHstore, []interface{}{}).Return(mock.Anything, nil)
```
@evanphx This mock is giving me trouble. Ideas?

```
panic: 

mock: Unexpected Method Call
-----------------------------

Exec(string,[]interface {})
        0: 
CREATE EXTENSION hstore

        1: []

The closest call I have is: 

Exec(string,[]interface {})
        0: 
CREATE EXTENSION hstore

        1: []
```

evanphx replied:
Ug, mocks. I wonder if this is the difference between an empty slice and a nil slice. Try changing 28 to `[]interface{}(nil)` or just remove that param all together.

jlsuttles replied:
Removing that param all together did not work, but using `[]interface{}(nil)` solves that problem. Yay :confetti_ball:

jlsuttles committed:
Remove some unused code

jlsuttles committed:
Use nil slice []interface{}(nil)

evanphx commented:
Woop!

jlsuttles committed:
Trying to get the mocks to play nice

Before these changes getting:

    panic: interface conversion: string is not sql.Result: missing method LastInsertId

After these changes getting:

    panic: interface conversion: interface is nil, not sql.Result

jlsuttles committed:
Yay I conquered the mocks!

jlsuttles committed:
Ping the db to see if tests should be skipped

jlsuttles committed:
Move database specifics for test to ENV vars

jlsuttles committed:
Create db table and extension only if they don't exist

jlsuttles committed:
Programmatically test db setup goes ok

jlsuttles committed:
Programmatically test pg db receive goes ok
